students/drovdahl/lesson07/html_render.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+Ryan Drovdahl's version
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    # 'html' is the default tag that is only used for testing
+    tag = 'html'
+    close_tag_line = True
+    link_tag = False
+    meta_tag = False
+
+    def __init__(self, content=None, style=None, id=None, indent=4):
```

Comment:
Still not as expected. Should receive `**kwargs` instead of fixed `style=..., id=...`. See https://github.com/UWPCE-PythonCert-ClassRepos/Wi2018-Classroom/blob/master/solutions/Session07/step_8/html_render.py

Reply:
I haven't had time to refactor this yet.  I have every intention of revisiting this though.  I spent a number of hours trying to make it work with **kwargs the first time and eventually gave up.  Given enough time however, I'd like to try and fix it.